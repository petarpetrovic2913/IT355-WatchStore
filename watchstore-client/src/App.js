import React, { Component } from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import { Provider } from 'react-redux';
import { toast } from 'react-toastify';
import './App.css';
import Layout from './layout/Layout';
import HomePage from './pages/HomePage/HomePage';
import ProductDetails from './pages/ProductDetails/ProductDetails';
import Register from './pages/Register/Register';
import AddProduct from './pages/AddProduct/AddProduct';
import UpdateProduct from './pages/UpdateProduct/UpdateProduct';
import store from './store';
import Customers from './pages/Customers/Customers';
import Login from './pages/Login/Login';
import jwt_decode from 'jwt-decode';
import { logout } from './store/actions/authActions';
import setJWTToken from './securityUtils/setJWTToken';
import { SET_CURRENT_USER } from './store/actions/actionTypes';
import Cart from './pages/Cart/Cart';
import UserSecuredRoute from './securityUtils/UserSecuredRoute';
import AdminSecuredRoute from './securityUtils/AdminSecureRoute';

const jwtToken = localStorage.jwtToken;

if (jwtToken) {
  setJWTToken(jwtToken);
  const decodedToken = jwt_decode(jwtToken);
  store.dispatch({
    type: SET_CURRENT_USER,
    payload: decodedToken
  });
  const currentTime = Date.now() / 1000;
  // handle tho logout
  if (decodedToken.exp < currentTime) {
    store.dispatch(logout());
    window.location.href('/');
  }
}

toast.configure();

class App extends Component {
  render() {
    return (
      <Provider store={store}>
        <Router>
          <Layout>
            {
              // Public Routes
            }
            <Route exact path="/" component={HomePage} />
            <Route exact path="/details/:id" component={ProductDetails} />
            <Route exact path="/register" component={Register} />
            <Route exact path="/login" component={Login} />

            {
              // Private Routes
            }

            <Switch>
              <UserSecuredRoute exact path="/cart" component={Cart} />
              <AdminSecuredRoute
                exact
                path="/customers"
                component={Customers}
              />
              <AdminSecuredRoute exact path="/add" component={AddProduct} />
              <AdminSecuredRoute
                exact
                path="/update/:id"
                component={UpdateProduct}
              />
            </Switch>
          </Layout>
        </Router>
      </Provider>
    );
  }
}

export default App;
