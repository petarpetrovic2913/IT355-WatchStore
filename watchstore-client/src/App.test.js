import React from 'react';
import { render } from '@testing-library/react';
import App from './App';

describe('Test App component', () => {
  it('renders without crashing', () => {
    const { container } = render(<App />);

    expect(container).toMatchSnapshot();
  });
});
