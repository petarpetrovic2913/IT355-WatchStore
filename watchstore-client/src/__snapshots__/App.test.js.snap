// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App component renders without crashing 1`] = `
<div>
  <div
    class="layoutHolder"
  >
    <div
      class="headerHolder "
    >
      <div
        class="links"
      >
        <a
          href="/register"
        >
          <p
            class="labelLink"
          >
            Sign up
          </p>
        </a>
        <a
          href="/login"
        >
          <p
            class="labelLink"
          >
            Login
          </p>
        </a>
      </div>
    </div>
    <div
      class="sidebarWrapper"
    >
      <div
        class="logoWrapper"
      >
        <a
          class="logoLink"
          href="/"
        >
          <img
            alt="AppLogo"
            src="slika1.png"
          />
        </a>
      </div>
      <form
        class="ui form form"
      >
        <div
          class="field searchWrapper"
        >
          <p
            class="label"
          >
            Price Down
          </p>
          <input
            name="priceDown"
            type="text"
            value=""
          />
        </div>
        <div
          class="field"
        >
          <p
            class="label"
          >
            Price Upper
          </p>
          <input
            name="priceUpper"
            type="text"
            value=""
          />
        </div>
        <div
          class="field"
        >
          <p
            class="label"
          >
            Unit in stock
          </p>
          <input
            name="unitInStock"
            type="text"
            value=""
          />
        </div>
        <div
          class="field submitButton"
        >
          <button
            class="ui button"
            type="submit"
          >
            Submit
          </button>
        </div>
      </form>
    </div>
    <div
      class="contentHolder"
    >
      <div
        class="ui icon input inputSearch"
      >
        <input
          placeholder="Search..."
          type="text"
        />
        <i
          aria-hidden="true"
          class="search circular link icon"
        />
      </div>
      
      <div>
        <button
          class="ui mini icon primary left labeled button pageNumButton"
        >
          <i
            aria-hidden="true"
            class="left chevron icon"
          />
          Previous
        </button>
        <span
          class="pageNum"
        >
          0
        </span>
        <button
          class="ui mini icon primary right labeled button pageNumButton"
        >
          <i
            aria-hidden="true"
            class="right chevron icon"
          />
          Next
        </button>
        <table
          class="ui table"
        >
          <thead
            class=""
          >
            <tr
              class=""
            >
              <th
                class=""
              >
                Model
              </th>
              <th
                class=""
              >
                Category
              </th>
              <th
                class=""
              >
                Description
              </th>
              <th
                class=""
              >
                Price
              </th>
              <th
                class=""
              >
                Unit in stock
              </th>
              <th
                class=""
              />
              <th
                class=""
              />
              <th
                class=""
              />
              <th
                class=""
              />
              <th
                class=""
              />
            </tr>
          </thead>
          <tbody
            class=""
          />
        </table>
        <button
          class="ui mini icon primary left labeled button pageNumButton"
        >
          <i
            aria-hidden="true"
            class="left chevron icon"
          />
          Previous
        </button>
        <span
          class="pageNum"
        >
          0
        </span>
        <button
          class="ui mini icon primary right labeled button pageNumButton"
        >
          <i
            aria-hidden="true"
            class="right chevron icon"
          />
          Next
        </button>
      </div>
    </div>
  </div>
</div>
`;
